tf_encrypted/tensor/int100.py

```diff
@@ -16,699 +22,719 @@
 from ..operations import secure_random
 
 
-def crt_factory(INT_TYPE, MODULI):
-
-    MATMUL_THRESHOLD = 1024
-
-    MODULUS = prod(MODULI)
-    BITSIZE = math.ceil(math.log2(MODULUS))
-
-    # make sure we have room for lazy reductions:
-    # - 1 multiplication followed by 1024 additions
-    for mi in MODULI:
-        assert 2 * math.log2(mi) + math.log2(1024) < math.log2(INT_TYPE.max)
-
-    #
-    # methods benefitting from precomputation
-    #
-
-    def gen_crt_recombine_lagrange():
-
-        # precomputation
-        n = [MODULUS // mi for mi in MODULI]
-        lambdas = [ni * inverse(ni, mi) % MODULUS for ni, mi in zip(n, MODULI)]
+def crt_factory(int_type, moduli):
+  """Chinese remainder theorem tensor factory."""
 
-        def crt_recombine_lagrange(x):
+  matmul_threshold = 1024
```

Comment:
i'm not sure this pattern of hiding that it's a constant is useful; is it because of something like this https://stackoverflow.com/questions/25184097/pylint-invalid-constant-name ?

Reply:
yes -- I think the solution is to not wrap these in functions, since there's really no need. then all of these would be treated as constants.

Reply:
treated as constants by linting? in the end I'm concerned by someone mistaken them for variables and changing their value. not sure python has any way of preventing this besides naming convention.

Reply:
yes, in which case we can use the upper case naming.  we can disable that pylint rule here in the meantime if that's preferable!

Reply:
I think that's better tbh, otherwise we're throwing information away. Agree on finding a better solution mid-long term.

Reply:
fixed in latest push!